test/config.js

```diff
@@ -482,6 +483,41 @@ captureBodyTests.forEach(function (captureBodyTest) {
   })
 })
 
+var usePathAsTransactionNameTests = [
+  { value: true, url: '/foo/bar?baz=2', transactionName: 'GET /foo/bar' },
+  { value: false, url: '/foo/bar?baz=2', transactionName: 'GET unknown route' }
+]
+
+usePathAsTransactionNameTests.forEach(function (usePathAsTransactionNameTest) {
+  test('usePathAsTransactionName => ' + usePathAsTransactionNameTest.value, function (t) {
+    t.plan(2)
+
+    var agent = Agent()
+    agent.start({
+      serviceName: 'test',
+      usePathAsTransactionName: usePathAsTransactionNameTest.value
+    })
+
+    var sendTransaction = agent._transport.sendTransaction
+    agent._transport.sendTransaction = function (trans, cb) {
+      t.ok(trans)
+      t.equal(trans.name, usePathAsTransactionNameTest.transactionName)
+      if (cb) process.nextTick(cb)
+    }
+    t.on('end', function () {
+      agent._transport.sendTransaction = sendTransaction
+    })
+
+    var req = new IncomingMessage()
+    req.url = usePathAsTransactionNameTest.url
+    req.method = 'GET'
+
+    var trans = agent.startTransaction()
+    trans.req = req
+    trans.end()
+ })
```

Comment:
I think our linter will complain about the indentation here. I should be two spaces :)

Reply:
they all stuck here, maybe something wrong in my code?
![image](https://user-images.githubusercontent.com/14990734/54702037-10fe4e00-4b71-11e9-8405-c371dfa1da47.png)

Reply:
Interesting 🤔 I'll take a look

Reply:
It was because `IncomingMessage` expected to have a `socket` property. When it didn't have that it threw an exception. But that exception was swallowed because `captureExceptions` wasn't set to `false`. After I did that I could see what the problem was:

```
/Users/watson/code/node_modules/elastic-apm-node/lib/parsers.js:99
      remote_address: req.socket.remoteAddress,
                                 ^

TypeError: Cannot read property 'remoteAddress' of undefined
    at Object.exports.getContextFromRequest (/Users/watson/code/node_modules/elastic-apm-node/lib/parsers.js:99:34)
    at Transaction.toJSON (/Users/watson/code/node_modules/elastic-apm-node/lib/instrumentation/transaction.js:137:41)
    at Transaction._encode (/Users/watson/code/node_modules/elastic-apm-node/lib/instrumentation/transaction.js:153:15)
    at Instrumentation.addEndedTransaction (/Users/watson/code/node_modules/elastic-apm-node/lib/instrumentation/index.js:112:65)
    at Instrumentation.wrappedAddEndedTransaction [as addEndedTransaction] (/Users/watson/code/node_modules/elastic-apm-node/lib/instrumentation/async-hooks.js:53:34)
    at Transaction.end (/Users/watson/code/node_modules/elastic-apm-node/lib/instrumentation/transaction.js:217:32)
    at Test.<anonymous> (/Users/watson/code/node_modules/elastic-apm-node/test/config.js:518:11)
    at Test.bound [as _cb] (/Users/watson/code/node_modules/elastic-apm-node/node_modules/tape/lib/test.js:76:32)
    at Test.run (/Users/watson/code/node_modules/elastic-apm-node/node_modules/tape/lib/test.js:95:10)
    at Test.bound [as run] (/Users/watson/code/node_modules/elastic-apm-node/node_modules/tape/lib/test.js:76:32)
    at Immediate.next (/Users/watson/code/node_modules/elastic-apm-node/node_modules/tape/lib/results.js:71:15)
    at processImmediate (timers.js:637:19)
```

I took the liberty to push a fix to your branch: https://github.com/elastic/apm-agent-nodejs/pull/907/commits/4d3e23c8f01e118b310af6dfe54224e0daa4aecb

Reply:
I see, thanks for your help 👍